autonomi/src/client/high_level/data/stream.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::client::data_types::chunk::ChunkAddress;
+use xor_name::XorName;
+
+use crate::Bytes;
+use crate::Client;
+use crate::client::GetError;
+
+type ChunkFetcher =
+    Box<dyn Fn(&[(usize, XorName)]) -> self_encryption::Result<Vec<(usize, Bytes)>> + Send + Sync>;
+
+pub struct DataStream {
+    streaming_decrypt: self_encryption::DecryptionStream<ChunkFetcher>,
+    finished: bool,
```

Comment:
oh, it's bit weird that the `wrapper` requires an additional `finished` flag.
maybe shall make `DecryptionStream` provide such status indicating getter ?
and seems if we had such getter in `DecryptionStream` , we then no need this `DataStream ` wrapper ?